packages/amplication-code-gen-types/src/code-gen-types.ts

```diff
@@ -225,3 +226,24 @@ export type PrismaDataSource = {
   provider: DataSourceProvider;
   urlEnv: string;
 };
+
+export interface ResolverMapping {
+  RESOLVER: ASTNode | undefined;
+  RESOLVER_BASE: ASTNode | undefined;
+  SERVICE: ASTNode | undefined;
+  ENTITY: ASTNode | undefined;
+  ENTITY_NAME: ASTNode | undefined;
+  ENTITY_QUERY: ASTNode | undefined;
+  ENTITIES_QUERY: ASTNode | undefined;
+  META_QUERY: ASTNode | undefined;
+  CREATE_MUTATION: ASTNode | undefined;
+  UPDATE_MUTATION: ASTNode | undefined;
+  DELETE_MUTATION: ASTNode | undefined;
+  CREATE_ARGS: ASTNode | undefined;
+  UPDATE_ARGS: ASTNode | undefined;
+  DELETE_ARGS: ASTNode | undefined;
+  FIND_MANY_ARGS: ASTNode | undefined;
+  FIND_ONE_ARGS: ASTNode | undefined;
+  CREATE_DATA_MAPPING: ASTNode | undefined;
+  UPDATE_DATA_MAPPING: ASTNode | undefined;
+}
```

Comment:
Instead of using this type, we want to use a generic type in order to allow the plugin writer to add a mapping.

The event parameter should be identical to the other events like CreateEntityController, and we should use 

templateMapping: { [key: string]: any };

Reply:
I tried to avoid that, cause the inner function is expecting this specific map. if the user will add whatever he wants, it will fail

Reply:
we must use a generic type, otherwise the plugin writer will not be able to do anything 
please copy the behavior from 
have a look at this issue with the draft for the doc

https://github.com/amplication/amplication/issues/4097